app/Http/Controllers/AntelopeActivity.php

```diff
@@ -57,30 +67,67 @@ protected function validator(array $data)
             'patrol_area' => ['required', 'array', 'min:1'],
             'patrol_area.*' => ['required', 'string'],
             'patrol_priorities' => ['required', 'integer'],
+            'flag' => ['required', 'boolean'],
+            'flag_reason' => ['string']
         ]);
     }
 
     /**
      * Create a new log instance after validation
      *
-     * @param  array  $data
+     * @param array $data
      * @return \App\User
+     * @throws \Exception
      */
     protected function create(array $data)
     {
-
         $data = $this->convertTimezone($data);
 
+        if ($data['flag'] === "true") {
+            $data['flag'] = true;
+        } else if ($data['flag'] === "false") {
+            $data['flag'] = false;
+        }
+
+        if (is_null($data['flag_reason'])) {
+            $data['flag_reason'] = "";
+        }
+
+        $now = new DateTime();
+        $start = new DateTime($data['patrol_start_date'] . 'T' . $data['start_time']);
+        $end = new DateTime($data['patrol_end_date'] . 'T' . $data['end_time']);
+        $diff = $end->diff($start);
+        $hours = $diff->h;
```

Comment:
I'd really look into using Carbon here, I'm not sure if it'd be easier for you or not, but just as a pointer.

Reply:
Ended up using Carbon to get the current user time.